lib/modules/manager/pep621/processors/pdm.ts

```diff
@@ -39,4 +48,75 @@ export class PdmProcessor implements PyProjectProcessor {
 
     return deps;
   }
+
+  async updateArtifacts(
+    updateArtifact: UpdateArtifact
+  ): Promise<UpdateArtifactsResult[] | null> {
+    const { config, updatedDeps, packageFileName } = updateArtifact;
+
+    const isLockFileMaintenance = config.updateType === 'lockFileMaintenance';
+
+    // abort if no lockfile is defined
+    const lockFileName = getSiblingFileName(packageFileName, 'pdm.lock');
+    try {
+      const existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
+      if (is.nullOrUndefined(existingLockFileContent)) {
+        logger.debug('No pdm.lock found');
+        return null;
+      }
+
+      const toolConstraint: ToolConstraint = {
+        toolName: 'pdm',
+        constraint: config.constraints?.pdm,
+      };
+
+      const execOptions: ExecOptions = {
+        docker: {},
+        toolConstraints: [toolConstraint],
+      };
+
+      // on lockFileMaintenance do not specify any packages and update the complete lock file
```

Comment:
Does `pdm update` update the complete lock file? In some languages we need to delete the lock file first to get it to do a proper full refresh

Reply:
Based on the documentation yes and during testing I have found nothing which contradicts that. 
https://pdm.fming.dev/2.6/reference/cli/#update_2